### PR TITLE
Untangling: Add Advanced Tools -> Database

### PR DESCRIPTION
**Pull request by fushar**
Fixes:

- https://github.com/Automattic/dotcom-forge/issues/9620

## Proposed Changes

This PR adds the Advanced Tools -> Database subtab under the `untangling/hosting-menu` flag.

image

## Why are these changes being made?

pbxlJb-6ye-p2

## Testing Instructions

1. Go to /sites
2. Click an Atomic site
3. Go to Advanced Tools -> Database subtab.
4. Verify that this tool is working.
5. You can also test on a site which does not allow "advanced hosting features" such as SFTP and Database by creating an Ecommerce trial site from https://wordpress.com/ecommerce/:
    - Verify that you can't see this feature both in Advanced Tools and in the original Server Settings.
7. Regression testing:
    - Go to Server Settings
    - Verify that the original `Database access` card is still working.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/sites/features.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export function useAreAdvancedHostingFeaturesSupported() {
+	const site = useSelector( getSelectedSite );
+	const features = useSelectedSiteSelector( getSiteFeatures );
+	const hasSftpFeature = useSelectedSiteSelector( siteHasFeature, FEATURE_SFTP );
+
+	if ( ! features ) {
+		return null;
+	}
+	return areHostingFeaturesSupported( site ) && hasSftpFeature;
```
Comment (fushar) [inline client/sites/features.ts:24]: The definition comes from

https://github.com/Automattic/wp-calypso/blob/0c5b372d9177d6d673f19524e53cabd7066c769c/client/hosting/server-settings/main.tsx#L275

**Review by taipeicoder — APPROVED** [review]
🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16990088

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @fushar for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16990088).

**Pull request merged**